src/parse.ts

```diff
@@ -0,0 +1,40 @@
+import type { VALIDATORS, ValidatorToType } from './validators';
+import { NUMBER_VALIDATOR, DATE_VALIDATOR, STRING_VALIDATOR } from './validators';
+
+const simplifiedISODateStringRegex = /^[+-]?\d{4}/;
+export const isISODateString = (value: unknown) =>
+  simplifiedISODateStringRegex.test(value as string)
+    ? !Number.isNaN(Number(new Date('1970-01-01T00:00:00.000Z')))
+    : false;
+
+export const parse = <V extends VALIDATORS>(validator: V) => (
+  value: unknown,
+): V extends keyof ValidatorToType ? ValidatorToType[V] : unknown => {
+  switch (validator) {
+    case NUMBER_VALIDATOR:
+      if (typeof value === 'string') {
+        if (value.trim() === '') {
+          return value as V extends keyof ValidatorToType ? ValidatorToType[V] : unknown;
+        }
+        const parsedNumber = Number(value);
+        return parsedNumber as V extends keyof ValidatorToType ? ValidatorToType[V] : unknown;
```

Comment:
Are these type assertions needed?

Reply:
They can be deleted if we change the return type of this function to ` V extends keyof ValidatorToType ? ValidatorToType[V] : unknown | unknown`.